gix-macros/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! A crate of useful macros used in `gix` primarily.
+//!
+//! Note that within `gix-*` crates, monomorphization should never be used for convenience, but only for performance
+//! reasons. And in the latter case, manual denomophization should be considered if the trait in questions isn't called
+//! often enough or measurements indicate that `&dyn Trait` is increasing the runtime. Thus, `gix-*` crates should probably
+//! by default prefer using `&dyn` unless measurements indicate otherwise.
+use proc_macro::TokenStream;
+
+/// When applied to functions or methods, it will turn it into a wrapper that will immediately call
+/// a de-monomorphized implementation (i.e. one that uses `&dyn Trait`).
+///
+/// That way, the landing-pads for convenience will be as small as possible which then delegate to a single
+/// function or method for implementation.
+///
+/// The parameters using the following traits can be de-monomorphized:
+///
+/// * `Into`
+/// * `AsRef`
+/// * `AsMut`
+#[proc_macro_attribute]
+pub fn momo(_attrs: TokenStream, input: TokenStream) -> TokenStream {
+    //TODO: alternatively parse ImplItem::Method
```

Comment:
Can this TODO be removed? Otherwise I have a feeling it could stay there for long, and I don't know what it means so couldn't easily follow up either.

Reply:
Yes, I've removed that now.

Reply:
Thanks! Just to double-check: did you see my note about letting it fail if it couldn't apply any conversion?

Reply:
Thanks, I've read that comment but somehow missed that part.